source/features/git-checkout-pr.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import ClippyIcon from 'octicon/clippy.svg';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import {getCurrentBranch} from '../github-helpers';
+
+function handleMenuOpening({delegateTarget: dropdown}: delegate.Event): void {
+	dropdown.classList.add('rgh-git-checkout'); // Mark this as processed
+	const [, user, repository] = select<HTMLAnchorElement>('.commit-ref.head-ref a')!.pathname.split('/', 3);
+	const isLocalPr = select('.user-select-contain.head-ref a')!.childElementCount === 1;
+	const tabContainer = select('[action="/users/checkout-preference"]', dropdown)!.closest<HTMLElement>('tab-container')!;
+	tabContainer.style.minWidth = '370px';
+	select('.UnderlineNav-body', tabContainer)!.append(
+		<button
+			name="type"
+			type="button"
+			role="tab"
+			aria-selected="false"
+			className="UnderlineNav-item flex-1 btn-link"
+		>
+			Git Checkout
+		</button>
+	);
+
+	tabContainer.append(
+		<div hidden role="tabpanel" className="p-3">
+			<p className="text-gray text-small">
+				Run in your project repository
+			</p>
+			<div className="copyable-terminal">
+				<div className="copyable-terminal-button">
+					<clipboard-copy
+						className="btn btn-sm zeroclipboard-button"
+						role="button"
+						for="rgh-checkout-pr"
+						aria-label="Copy to clipboard"
+						data-copy-feedback="Copied!"
+					>
+						<ClippyIcon/>
+					</clipboard-copy>
+				</div>
+				<pre
+					id="rgh-checkout-pr"
+					className="copyable-terminal-content"
+				>
+					<span className="user-select-contain">
+						{isLocalPr || `git remote add ${user} ${location.origin}/${user}/${repository}.git -f -t ${getCurrentBranch()!}\n`}
```

Comment:
Sorry to interrupt so late again, but here only the HTTPS protocol is supported. I'd prefer the SSH protocol however, so that git doesn't ask for my credentials whenever I push.

See also the "Clone" button: HTTPS (`https://github.com/sindresorhus/refined-github.git`) vs. SSH (`git@github.com:sindresorhus/refined-github.git`)

![image](https://user-images.githubusercontent.com/202916/96732176-55d71000-13b8-11eb-92d0-cd0be00ee033.png)

Reply:
I was going to say that too but I kept my mouth shut not to make this more complicated

Reply:
> so that git doesn't ask for my credentials whenever I push.

That only happens because you haven’t set it up yet: https://docs.github.com/en/free-pro-team@latest/github/using-git/which-remote-url-should-i-use#cloning-with-https-urls

Reply:
@fregante I can't see how that page describes how to avoid being asked for credentials (username and password or personal access token) with HTTP(S) URLs. I've set it up locally using SSH URLs, thus the SSH keypair enables passwordless (or tokenless) authentication.

Reply:
It does:

> You can use a credential helper so Git will remember your GitHub credentials every time it talks to GitHub. For more information, see "Caching your GitHub credentials in Git."

I use HTTPS and I've only ever had to enter the credentials once, just like you put your public key on GitHub.com. It's a one-time setup, no one enters passwords every time they push.

Reply:
@yakov116 do you feel like reimplementing this by offering both options? 😄 

I think we could keep it simple and offer them one under the other, with an opening text like "Pick either one"


```
git remote add yakov116 https://github.com/yakov116/refined-github.git
git fetch yakov116 check-it-out
git switch --track yakov116/check-it-out
```

```
git remote add yakov116 git@github.com:yakov116/refined-github.git
git fetch yakov116 check-it-out
git switch --track yakov116/check-it-out
```

Reply:
> do you feel like reimplementing this by offering both options? 😄
If I can do it in the next 5 min 💯 

So 2 boxes not another tab like github?

Reply:
> You can use a credential helper so Git will remember your GitHub credentials every time it talks to GitHub.

@fregante Ah okay, I didn't know of the credential helper. Did you set the timeout to a really long time? Otherwise, it still asks more often than with the SSH-based approach :wink:

> no one enters passwords every time they push

I just thought that everybody uses SSH :sweat_smile:

Reply:
![image](https://user-images.githubusercontent.com/16872793/96797223-db18ef80-13cd-11eb-90bb-4759100344d0.png)


I can probably use a map to shrink the code in half... maybe later times up.

Reply:
A map?